TelemetryProcessors/SamplingTelemetryProcessor.ts

```diff
@@ -0,0 +1,45 @@
+import ContractsModule = require("../Library/Contracts");
+import { CorrelationContext } from "../AutoCollection/CorrelationContextManager";
+
+/**
+ *  A telemetry processor that handles sampling.
+ */
+export function samplingTelemetryProcessor(envelope: ContractsModule.Contracts.Envelope, contextObjects: { correlationContext: CorrelationContext }): boolean {
+    var samplingPercentage = envelope.sampleRate; // Set for us in Client.getEnvelope
+    var isSampledIn = false;
+
+    if (samplingPercentage === null || samplingPercentage === undefined || samplingPercentage >= 100) {
+        return true;
+    } else if (contextObjects.correlationContext && contextObjects.correlationContext.operation) {
+        // If we're using dependency correlation, sampling should retain all telemetry from a given request 
+        isSampledIn = getSamplingHashCode(contextObjects.correlationContext.operation.id) < samplingPercentage;
+    } else {
+        // If we're not using dependency correlation, sampling should use a random distribution on each item
+        isSampledIn = (Math.random() * 100) < samplingPercentage;
+    }
+
+    return isSampledIn;
+}
+
+/** Ported from AI .NET SDK */
+export function getSamplingHashCode(input: string): number {
```

Comment:
It's not used for basically historical reasons :) I had already written this together for Breeze and simply moved it across. When I wrote it the first time I didn't know JS had hashing logic at all. I believe they're basically the same however. A base SDK would seem useful for this...